### ISSUE TITLE
Invalid usage of Function(np.random.standard_normal) with parameters (int64)

### ISSUE THREAD

**Issue by seanlaw** [numba/numba#1596]
I'm trying out numba for the first time on a simple Black Scholes function:

    @numba.jit(nopython=True)
    def bsm(I=100000000):
        S0 = 100.0
        K = 105.0
        T = 1.0
        r = 0.05
        sigma = 0.2
    
        z = np.random.standard_normal(I)
        ST = S0 * np.exp((r - 0.5 * np.power(sigma, 2)) * T + sigma  * np.sqrt(T) * z)
        hT = np.maximum(ST - K, 0)
        C0 = np.exp(-r * T)*np.sum(hT)/I
    
        return C0

It's returning the following error:

    TypingError: Invalid usage of Function(np.random.standard_normal) with parameters (int64)
    Known signatures:
     * () -> float64
    File "<ipython-input-43-204a1a6e6eab>", line 9
    
    Failed at nopython (nopython frontend)
    Invalid usage of Function(np.random.standard_normal) with parameters (int64)
    Known signatures:
     * () -> float64
    File "<ipython-input-43-204a1a6e6eab>", line 9

It's not clear how I should interpret this or go about resolving this error. Any suggestions?

**Comment by seibert** [comment]
This error means that Numba does not yet support returning arrays from `standard_normal()` (or any of the `np.random` functions) at the moment.  Specifically, the error is saying that the only know type signature for `standard_normal` is `()`, which means no arguments, and a return value of type `float64`.

As a workaround until we do support the random array functionality, you can make a helper function:
``` python
@jit(nopython=True)
def standard_normal(size):
    r = np.empty(size, dtype=np.float64)
    for i in range(size):
        r[i] = np.random.standard_normal()
    return r
```
You can call this function instead of `np.random.standard_normal()` in `bsm()`.

**Comment by pitrou** [comment]
Fixed in PR #1758.
